kustomize/base/deployment.yaml

```diff
@@ -0,0 +1,62 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nodeless-cost-calculator
+  namespace: default
+  labels:
+    app.kubernetes.io/name: nodeless-cost-calculator
+spec:
+  selector:
+    matchLabels:
+      run: nodeless-cost-calculator
+  replicas: 1
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      namespace: default
+      labels:
+        run: nodeless-cost-calculator
+        app.kubernetes.io/name: nodeless-cost-calculator
+    spec:
+      serviceAccount: nodeless-cost-calculator
+      serviceAccountName: nodeless-cost-calculator
+      containers:
+        - name: nodeless-cost-calculator
+          image: elotl/nodeless-cost-calculator:v0.0.3-37-g535a896-dirty
+          ports:
+            - containerPort: 5000
+              protocol: TCP
+          env:
+            - name: CLOUD_PROVIDER
+              value: azure
+            - name: REGION
+              value: East US
+        - name: redis
+          image: redis
+          ports:
+            - containerPort: 6379
+              name: redis
+              protocol: TCP
+        - name: cloudinfo
+          image: banzaicloud/cloudinfo
+          args: ["cloudinfo", "--scrape=true", "--config=/etc/cloudinfo/config.toml",]
+          ports:
+            - containerPort: 8000
+              protocol: TCP
+          volumeMounts:
+            - name: cloudinfo-config
+              mountPath: /etc/cloudinfo/config.toml
+              subPath: config.toml
+              readOnly: true
+          readinessProbe:
+            httpGet:
+              port: 8000
+              path: /api/v1/providers/azure/services/compute/regions/eastus/products
+            initialDelaySeconds: 120
+            periodSeconds: 60
+            timeoutSeconds: 1200
```

Comment:
Some food for thought: here the three services are coupled together; if the deployment is changed or reconfigured, then the whole pod will restart which means all the data in redis will go away, the user will need to wait for cloudinfo to re-populate it, etc. If they are deployed as independent pods instead, then e.g. updating the cost calculator will not affect data in Redis, or trigger a re-scraping of all pricing info (which is time consuming). Probably a better UX if a user updates the app, or changes its configuration.

Reply:
That's a good point. My first take was to have it all as one pod, as initial idea was to share instance data as a file mounted in both containers, but this turn out to be not supported out of the box as data storage. With redis indeed it makes much more sense to have it separate.

Reply:
addressed it in separate PR: https://github.com/elotl/nodeless-cost-calculator/pull/8 which is based on this branch